zad2/warriors.rb

```diff
@@ -1,12 +1,78 @@
-class Warrior
+class Character
+  
+  attr_accessor :name
+  attr_accessor :level
+
   def initialize(name:, level:)
+    self.name=name
+    self.level=level
   end
 
   def strength
+    [*1..12].sample + self.level
+  end
+
+  def card
+    puts"Name: #{name}, Level: #{level}"
   end
 end
 
+class Warrior < Character
+end
+class Monster < Character
+end
+
 class BattleArena
+  
+  attr_accessor :first_character
+  attr_accessor :second_character
+
   def initialize(first_character, second_character)
+    self.first_character=first_character
+    self.second_character=second_character
+  end
+
+  def fight(winner,loser)
+    if winner.is_a? Warrior 
+      if winner.level < loser.level
+        winner.level += loser.level-winner.level + 1
+      elsif winner.level == loser.level 
+        winner.level + 1
+      end
+    end
+  end
+
+  def announcement(damage1,damage2)
+    puts "#{first_character.name} attacked #{second_character.name} with #{damage1} damage"
+    puts "#{second_character.name} attacked #{first_character.name} with #{damage2} damage"
+  end
+
+  def who_win(damage1,damage2)
```

Comment:
nazwa metody wskazuje na to, że zostanie zwrócony obiekt typu `Character` (albo nil w przypadku remisu - co się dzieje)